balancer/endpointsharding/endpointsharding.go

```diff
@@ -76,24 +75,28 @@ type endpointSharding struct {
 	mu sync.Mutex // Sync updateState callouts and childState recent state updates
 }
 
-// UpdateClientConnState creates a child for new endpoints and deletes children
-// for endpoints that are no longer present. It also updates all the children,
-// and sends a single synchronous update of the childrens' aggregated state at
-// the end of the UpdateClientConnState operation. If any endpoint has no
-// addresses, returns error without forwarding any updates. Otherwise returns
-// first error found from a child, but fully processes the new update.
-func (es *endpointSharding) UpdateClientConnState(state balancer.ClientConnState) error {
-	if len(state.ResolverState.Endpoints) == 0 {
+// ValidateEndpoints returns an error if the endpoints list is empty, or no
+// addresses are present in endpoint list.
+func ValidateEndpoints(endpoints []resolver.Endpoint) error {
+	if len(endpoints) == 0 {
 		return errors.New("endpoints list is empty")
 	}
-	// Check/return early if any endpoints have no addresses.
-	// TODO: make this configurable if needed.
-	for i, endpoint := range state.ResolverState.Endpoints {
-		if len(endpoint.Addresses) == 0 {
-			return fmt.Errorf("endpoint %d has empty addresses", i)
+
+	for _, endpoint := range endpoints {
+		for range endpoint.Addresses {
+			return nil
 		}
 	}
+	return errors.New("endpoints list contains no addresses")
+}
 
+// UpdateClientConnState creates a child for new endpoints and deletes children
+// for endpoints that are no longer present. It also updates all the children,
+// and sends a single synchronous update of the childrens' aggregated state at
+// the end of the UpdateClientConnState operation. If any endpoint has no
+// addresses it will ignore that endpoint. Otherwise, returns first error found
+// from a child, but fully processes the new update.
```

Comment:
Nobody will see this godoc comment here since it's on an unexported type.

Reply:
Are you ok with that?

Reply:
I'm fine with comments that aren't visible to anyone but someone reading the code.  But make sure they aren't supposed to be showing up in godoc.  Godoc should have enough info that the package is usable by someone who doesn't know what it does before viewing the godocs, and unexported comments should focus on how the implementation works and why it's the way it is.

Reply:
I reread this comment and I think the language here appropriate for this not public facing operation (all relate to the UpdateClientConnState() operation), and the exported comments seem good enough to me. Do you want me to take some of this information and put it in exported go doc?